c++/src/capnp/compat/byte-stream.c++

```diff
@@ -1014,10 +1005,10 @@ private:
       left.back() = pieces[splitPiece].slice(0, splitByte);
       right.front() = pieces[splitPiece].slice(splitByte, pieces[splitPiece].size());
 
-      return SplitResult {
-        .piecesToWrite = kj::mv(left),
-        .rest = kj::mv(right)
-      };
+      return writeFirstPieces(left).attach(kj::mv(left))
+          .then([this,right=kj::mv(right),limit,writeFirstPieces]() mutable {
+        return splitAndWrite(right, limit, writeFirstPieces).attach(kj::mv(right));
```

Comment:
I think this should call `write()`, not `splitAndWrite()`. In the case of a borrowed stream, the limit was a one-time thing...

Reply:
Yep, good catch, thanks.